ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsInitializeIngestion.scala

```diff
@@ -173,4 +173,87 @@ private[backend] trait StorageBackendTestsInitializeIngestion
     metering2.applicationData should have size 2 // Partially ingested data removed
 
   }
+
+  it should "delete overspill entries written before first ledger end update" in {
+    val signatory = Ref.Party.assertFromString("signatory")
+
+    val dtos1: Vector[DbDto] = Vector(
+      // 1: config change
+      dtoConfiguration(offset(1), someConfiguration),
+      // 2: party allocation
+      dtoPartyEntry(offset(2), "party1"),
+      // 3: package upload
+      dtoPackage(offset(3)),
+      dtoPackageEntry(offset(3)),
+      // 4: transaction with create node
+      dtoCreate(offset(4), 1L, hashCid("#4"), signatory = signatory),
+      DbDto.CreateFilter(1L, someTemplateId.toString, someParty.toString),
+      dtoCompletion(offset(4)),
+      // 5: transaction with exercise node and retroactive divulgence
+      dtoExercise(offset(5), 2L, false, hashCid("#4")),
+      dtoDivulgence(Some(offset(5)), 3L, hashCid("#4")),
+      dtoCompletion(offset(5)),
+      // Transaction Metering
+      dtoMetering("AppA", offset(1)),
+      dtoMetering("AppB", offset(4)),
+    )
+
+    val readers = Set(signatory)
+
+    // Initialize
+    executeSql(backend.parameter.initializeParameters(someIdentityParams))
+    executeSql(backend.meteringParameter.initializeLedgerMeteringEnd(someLedgerMeteringEnd))
+
+    // Start the indexer (a no-op in this case)
+    val end1 = executeSql(backend.parameter.ledgerEnd)
+    executeSql(backend.ingestion.deletePartiallyIngestedData(end1))
+
+    // Insert first batch of updates, but crash before writing the first ledger end
+    executeSql(ingest(dtos1, _))
```

Comment:
What causes the crash in this test?

Reply:
It's a simulated crash. Normally, the indexer calls `ingest` and then `updateLedgerEnd`. If the indexer crashes in between those two calls, the data written by `ingest` needs to be deleted.